libs/agno/agno/workflow/v2/task.py

```diff
@@ -0,0 +1,340 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Iterator, List, Optional, Type, Union
+from uuid import uuid4
+
+from agno.agent import Agent
+from agno.media import AudioArtifact, ImageArtifact, VideoArtifact
+from agno.run.response import RunResponse
+from agno.run.team import TeamRunResponse
+from agno.run.workflow import WorkflowRunEvent, WorkflowRunResponse
+from agno.team import Team
+from agno.utils.log import logger
+
+
+@dataclass
+class TaskInput:
+    """Input data for a task execution"""
+
+    query: Optional[str] = None
+    message: Optional[str] = None
+
+    # Context and state
+    context: Optional[Dict[str, Any]] = None
+    workflow_session_state: Optional[Dict[str, Any]] = None
+
+    # Previous task outputs (for chaining)
+    previous_outputs: Optional[Dict[str, Any]] = None
+
+    # Media inputs
+    images: Optional[List[ImageArtifact]] = None
+    videos: Optional[List[VideoArtifact]] = None
+    audio: Optional[List[AudioArtifact]] = None
+
+    def get_primary_input(self) -> str:
+        """Get the primary text input (query or message)"""
+        return self.query or self.message or ""
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Convert to dictionary"""
+        return {
+            "query": self.query,
+            "message": self.message,
+            "context": self.context,
+            "workflow_session_state": self.workflow_session_state,
+            "previous_outputs": self.previous_outputs,
+            "images": [img.to_dict() for img in self.images] if self.images else None,
+            "videos": [vid.to_dict() for vid in self.videos] if self.videos else None,
+            "audio": [aud.to_dict() for aud in self.audio] if self.audio else None,
+            "extra_data": self.extra_data,
+        }
+
+
+@dataclass
+class TaskOutput:
+    """Output data from a task execution"""
+
+    # Primary output
+    content: Optional[str] = None
+
+    # Execution response
+    response: Optional[Union[RunResponse, TeamRunResponse]] = None
+
+    # Media outputs
+    images: Optional[List[ImageArtifact]] = None
+    videos: Optional[List[VideoArtifact]] = None
+    audio: Optional[List[AudioArtifact]] = None
+
+    # Structured data
+    data: Optional[Dict[str, Any]] = None
+
+    # Execution metadata
+    metadata: Optional[Dict[str, Any]] = None
```

Comment:
lets rather unpack this into the fields. Strcture is good here I think

Reply:
do you mean to have the keys of meta data has individual field values?
```py
        # Create metadata
        metadata = {
            "task_name": self.name,
            "task_id": self.task_id,
            "executor_type": self._executor_type,
            "executor_name": self.executor_name,
        }
```